### PR TITLE
Cvsl 1911 fix sorting and review needed issues

### PR DESCRIPTION
**Pull request by justin-gardiner**

### CODE REVIEW

File: server/routes/support/handlers/probationTeam.ts [hunk 1]
```diff
@@ -21,16 +22,22 @@ export default class ProbationTeamRoutes {
...
-          comCase.hardStopWarningDate && format(parseCvlDate(comCase.hardStopWarningDate), 'dd/MM/yyyy'),
-      }
-    })
+    )
+      .map(comCase => {
+        const releaseDate = comCase.releaseDate ? format(parseCvlDate(comCase.releaseDate), 'dd MMM yyyy') : 'not found'
+        return {
+          ...comCase,
+          releaseDate,
+          licenceStatus: comCase.isReviewNeeded ? LicenceStatus.REVIEW_NEEDED : comCase.licenceStatus,
```
Comment (SebNegahbanMoJ) [inline server/routes/support/handlers/probationTeam.ts:31]: Is this not also needed for the `probationStaff` vary caseload?

File: server/routes/support/handlers/probationTeam.ts [hunk 2]
```diff
@@ -21,16 +22,22 @@ export default class ProbationTeamRoutes {
...
+          sortDate: comCase.releaseDate && parseCvlDate(comCase.releaseDate),
+          hardStopDate: comCase.hardStopDate && format(parseCvlDate(comCase.hardStopDate), 'dd/MM/yyyy'),
+          hardStopWarningDate:
+            comCase.hardStopWarningDate && format(parseCvlDate(comCase.hardStopWarningDate), 'dd/MM/yyyy'),
+        }
+      })
+      .sort((a, b) => {
+        return (a.sortDate?.getTime() || 0) - (b.sortDate?.getTime() || 0)
```
Comment (SebNegahbanMoJ) [inline server/routes/support/handlers/probationTeam.ts:39]: **thought (non-blocking):** I don't think it matters with the number of cases we're dealing with, but I wonder if there's an efficiency difference between creating a `releaseDateString` that is the formatted `releaseDate`, then comparing `releaseDate` directly here.

File: server/routes/support/handlers/probationTeam.ts (hunk 1 above)
Comment (justin-gardiner) [reply server/routes/support/handlers/probationTeam.ts:31]: yes, it should be, added

File: server/routes/support/handlers/probationTeam.ts (hunk 2 above)
Comment (justin-gardiner) [reply server/routes/support/handlers/probationTeam.ts:39]: quite possibly, I guess it would be worth testing if it started to get slow.

**Review by SebNegahbanMoJ — APPROVED** [review]

**Pull request merged**